prow/plugins/dco/dco.go

```diff
@@ -0,0 +1,361 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package dco implements a DCO (https://developercertificate.org/) checker plugin
+package dco
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+
+	"regexp"
+
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/pluginhelp"
+	"k8s.io/test-infra/prow/plugins"
+)
+
+const (
+	pluginName               = "dco"
+	dcoContextName           = "dco"
+	dcoContextMessageFailed  = "Commits in PR missing Signed-off-by"
+	dcoContextMessageSuccess = "All commits have Signed-off-by"
+
+	dcoYesLabel        = "dco-signoff: yes"
+	dcoNoLabel         = "dco-signoff: no"
+	dcoMsgPruneMatch   = "Thanks for your pull request. Before we can look at it, you'll need to add a 'DCO signoff' to your commits."
+	dcoNotFoundMessage = `Thanks for your pull request. Before we can look at it, you'll need to add a 'DCO signoff' to your commits.
+
+:memo: **Please follow instructions in the [contributing guide](%s) to update your commits with the DCO**
+
+Full details of the Developer Certificate of Origin can be found at [developercertificate.org](https://developercertificate.org/).
+
+**The list of commits missing DCO signoff**:
+
+%s
+
+<details>
+
+%s
+</details>
+`
+)
+
+var (
+	checkDCORe = regexp.MustCompile(`(?mi)^/check-dco\s*$`)
+	testRe     = regexp.MustCompile(`(?mi)^signed-off-by:`)
+)
+
+func init() {
+	plugins.RegisterPullRequestHandler(pluginName, handlePullRequestEvent, helpProvider)
+	plugins.RegisterGenericCommentHandler(pluginName, handleCommentEvent, helpProvider)
+}
+
+func helpProvider(config *plugins.Configuration, enabledRepos []string) (*pluginhelp.PluginHelp, error) {
+	// The {WhoCanUse, Usage, Examples, Config} fields are omitted because this plugin cannot be
+	// manually triggered and is not configurable.
```

Comment:
This comment isn't quite right. Just the `Config` field is omitted. The other fields are present as part of the command.